frontend/src/components/Setup/Setup.tsx

```diff
@@ -15,134 +17,156 @@ type Props = {
-            if (!name || registered) return
-            event.preventDefault()
-            onDevice({ ...device, name })
-            setRegistering(true)
-          }}
-        >
-          <section className={css.device}>
-            <div className={css.name}>
-              <TextField
-                label="This System's Name"
-                className={css.input}
-                disabled={registering || registered}
-                value={name || device.name}
-                variant="filled"
-                error={!!nameError}
-                onChange={event => {
-                  const value = event.target.value.replace(REGEX_NAME_SAFE, '')
-                  if (value !== event.target.value) {
-                    setNameError('Device names can only contain alpha numeric characters.')
-                  } else if (nameBlacklist.includes(value.toLowerCase().trim())) {
-                    setNameError('That device name is already in use.')
-                    setDisableRegister(true)
-                  } else {
-                    setNameError(undefined)
-                    setDisableRegister(false)
-                  }
-                  setName(value)
-                }}
-                autoFocus={true}
-                onFocus={event => event.target.select()}
-                helperText={nameError || (!registered && '*Must be unique')}
-                inputProps={{ 'data-lpignore': 'true' }}
-              />
-              {deleting ? (
-                <CircularProgress className={css.loading} size={styles.fontSizes.lg} />
-              ) : (
-                device.uid && (
-                  <Tooltip title="Delete">
-                    <IconButton
-                      onClick={() => {
-                        if (window.confirm(confirmMessage)) {
-                          onDelete()
-                          setDeleting(true)
-                        }
-                      }}
-                    >
-                      <Icon name="trash-alt" size="md" />
-                    </IconButton>
-                  </Tooltip>
-                )
-              )}
-              {registered || (
-                <Button
-                  color="primary"
-                  variant="contained"
-                  size="medium"
-                  disabled={!name || disableRegister}
-                  type="submit"
-                >
-                  {registered ? 'Registered' : 'Register'}
-                  {registering ? (
-                    <CircularProgress className={css.registering} size={styles.fontSizes.lg} thickness={4} />
-                  ) : (
-                    <Icon name="check" weight="regular" inline />
-                  )}
-                </Button>
-              )}
-            </div>
-          </section>
-        </form>
-      </Body>
+    <>
+      <Container
+        header={
+          <>
+            <Breadcrumbs />
+            {registered && (
+              <>
```

Comment:
Wrapper tags seem redundant?

Reply:
Yes there is an extra one hanging around. I'll remove.